### PR TITLE
Manual backport of 51456 merge feature relocation

### PR DESCRIPTION
**Pull request by m-kuhn**
Backport https://github.com/qgis/QGIS/pull/51456

Currently the logic of the merge selected features tool is implemented completely in the app module. This logic has gained some complexity but is not CI tested.

This PR moves part of the logic to QgsVectorLayerEditUtils and add some tests for it. This will also make it accessible for python plugins.

------

The original pull request is shipped with 3.30.0 for a month and has been extensively tested by @andreasneumann and @ponceta

### CODE REVIEW

File: tests/src/python/test_qgsvectorlayereditutils.py [hunk 1]
```diff
@@ -266,6 +282,158 @@ def testAddRingV2SelectedFeatures(self):
...
+
+        self.assertTrue(vle.moveVertexV2(QgsPoint(3, 4, 5, 6), f2.id(), 0))
+        f2 = layer.getFeature(2)
+        self.assertTrue(f2.geometry().constGet().isMeasure())
+        self.assertTrue(f2.geometry().constGet().is3D())
+
+        self.assertEqual(f2.geometry().constGet().x(), 3)
+        self.assertEqual(f2.geometry().constGet().y(), 4)
+        self.assertEqual(f2.geometry().constGet().z(), 5)
+        self.assertEqual(f2.geometry().constGet().m(), 6)
```
Comment (domi4484) [inline tests/src/python/test_qgsvectorlayereditutils.py:435]: @m-kuhn all those "moveVertex" tests belongs in fact to another fix and are causing the test failures here.

File: tests/src/python/test_qgsvectorlayereditutils.py [hunk 2]
```diff
@@ -266,6 +282,158 @@ def testAddRingV2SelectedFeatures(self):
...
+
```
Comment (signedav) [inline tests/src/python/test_qgsvectorlayereditutils.py:436]:
```suggestion
```

Comment (signedav) [reply tests/src/python/test_qgsvectorlayereditutils.py:436]: @domi4484 As far as I understand you, this are tests added in this commit but concerning another implementation. So we just can remove it?

**Review by domi4484 — APPROVED** [review]

**Review by signedav — APPROVED** [review]

**Comment by nyalldawson** [comment]
Why was this merged into both the 3.28 AND queued ltr branch?

It's now impossible to rebase the queued_ltr_branch cleanly onto release_3-28. @m-kuhn can you please do this for me?

**Comment by signedav** [comment]
Merging into queued_ltr_branch was a mistake. It can be reverted there. This would solve the rebasing problem, right?

**Comment by agiudiceandrea** [comment]
@m-kuhn could you please see https://github.com/qgis/QGIS/issues/52738#issuecomment-1513079093?

**Pull request merged**